libr/lang/p/qjs.c

```diff
@@ -71,17 +87,43 @@ static void plugin_manager_add_core_plugin(QjsPluginManager *pm, const char *nam
 	}
 }
 
+static QjsIoPlugin *plugin_manager_add_io_plugin(QjsPluginManager *pm, const char *name, JSContext *ctx, RIOPlugin *iop, JSValue func) {
+	r_return_val_if_fail (pm, NULL);
+
+	QjsIoPlugin *cp = RVecIoPlugin_emplace_back (&pm->io_plugins);
+	if (cp) {
+		cp->name = name? strdup (name): NULL;
+		cp->ctx = ctx;
+		cp->iop = iop;
+		cp->fn_check_js = func;
+		// cp->qctx.open_func = func;
```

Comment:
not filled in yet? or are you trying to get MVP up and running first?

Reply:
its returning the object and filling the fields later, otherwise ill need to add over9000 parameters to this function and change the signature everytime i update the api to support more callbacks